apps/studio/src/components/Searchbar/SearchModal.tsx

```diff
@@ -0,0 +1,129 @@
+import { useState } from "react"
+import {
+  Modal,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Text,
+} from "@chakra-ui/react"
+import { Searchbar as OgpSearchBar } from "@opengovsg/design-system-react"
+import { useDebounce } from "@uidotdev/usehooks"
+
+import type { SearchResultResource } from "~/server/modules/resource/resource.types"
+import { useBanner } from "~/hooks/useBanner"
+import { trpc } from "~/utils/trpc"
+import { isMac } from "./isMac"
+import {
+  InitialState,
+  LoadingState,
+  NoResultsState,
+  SearchResultsState,
+} from "./SearchModalBodyContentStates"
+
+interface SearchModalProps {
+  isOpen: boolean
+  onClose: () => void
+  siteId: string
+}
+export const SearchModal = ({ siteId, isOpen, onClose }: SearchModalProps) => {
+  const banner = useBanner()
+  const mt = banner ? "3rem" : "0.5rem"
+  const [searchValue, setSearchValue] = useState("")
+  const debouncedSearchTerm = useDebounce(searchValue, 300)
+  const { data, isLoading } = trpc.resource.search.useInfiniteQuery({
+    siteId,
+    query: debouncedSearchTerm,
+  })
```

Comment:
issue: I think we might need to invalidate this query if the page title of any resource was changed, or if any resources were modified (i.e. blob, etc).

Reply:
IMO the likelihood of a _"search -> _(resource update)_ -> load more"_ is rather low

Plus, it's generally harmless at the moment to not do that since the query is not a "strict" one e.g. the order **must** follow a certain order - it's not an outright error if it really happen